rosetta/transactor/transactor.go

```diff
@@ -0,0 +1,592 @@
+	if sendSymbol != dps.FlowSymbol || receiveSymbol != dps.FlowSymbol {
+
+		return nil, failure.InvalidIntent{
+			Description: failure.NewDescription("invalid currencies found",
+				failure.WithString("sender", send.AccountID.Address),
+				failure.WithString("receiver", receive.AccountID.Address),
+				failure.WithString("withdrawal_currency", send.Amount.Currency.Symbol),
+				failure.WithString("deposit_currency", receive.Amount.Currency.Symbol)),
+		}
+	}
+
+	// Validate the sender and the receiver account IDs.
+	_, err = t.validate.Account(send.AccountID)
+	if err != nil {
+		return nil, fmt.Errorf("invalid sender account: %w", err)
+	}
+	_, err = t.validate.Account(receive.AccountID)
+	if err != nil {
+		return nil, fmt.Errorf("invalid receiver account: %w", err)
+	}
+
+	// Validate that the specified operations are transfers.
+	if send.Type != dps.OperationTransfer || receive.Type != dps.OperationTransfer {
+		return nil, failure.InvalidIntent{
+			Description: failure.NewDescription("only transfer operations are supported",
+				failure.WithString("withdrawal_type", send.Type),
+				failure.WithString("deposit_type", receive.Type),
+			),
+		}
+	}
+
+	amount := amounts[0]
+	if amount < 0 {
+		amount = -amount
+	}
+
+	intent := Intent{
+		From:     flow.HexToAddress(send.AccountID.Address),
+		To:       flow.HexToAddress(receive.AccountID.Address),
+		Amount:   cadence.UFix64(amount),
+		Payer:    flow.HexToAddress(send.AccountID.Address),
+		Proposer: flow.HexToAddress(send.AccountID.Address),
+		GasLimit: flow.DefaultMaxTransactionGasLimit,
+	}
+
+	return &intent, nil
+}
+
+// CompileTransaction creates a complete Flow transaction from the given intent and metadata.
+func (t *Transactor) CompileTransaction(rosBlockID identifier.Block, intent *Intent, sequence uint64) (string, error) {
+
+	// Generate script for the token transfer.
+	script, err := t.generate.TransferTokens(dps.FlowSymbol)
+	if err != nil {
+		return "", fmt.Errorf("could not generate transfer script: %w", err)
+	}
+
+	// TODO: Allow arbitrary proposal key index
+	// => https://github.com/optakt/flow-dps/issues/369
+
+	// Create the transaction.
+	unsignedTx := sdk.NewTransaction().
+		SetScript(script).
+		SetReferenceBlockID(sdk.HexToID(rosBlockID.Hash)).
+		SetPayer(sdk.Address(intent.Payer)).
+		SetProposalKey(sdk.Address(intent.Proposer), 0, sequence).
+		AddAuthorizer(sdk.Address(intent.From)).
+		SetGasLimit(intent.GasLimit)
+
+	receiver := cadence.NewAddress(flow.BytesToAddress(intent.To.Bytes()))
+
+	// Add the script arguments - the amount and the receiver.
+	_ = unsignedTx.AddArgument(intent.Amount)
+	_ = unsignedTx.AddArgument(receiver)
+
+	payload, err := t.encodeTransaction(unsignedTx)
+	if err != nil {
+		return "", fmt.Errorf("could not encode transaction: %w", err)
+	}
+
+	return payload, nil
+}
+
+func (t *Transactor) HashPayload(rosBlockID identifier.Block, unsigned string, signer identifier.Account) (string, error) {
```

Comment:
I think `SigningPayload` or something similar better conveys what's returned here.

Especially with the `hash` endpoint returning the ID of the transaction, but also we're assigning the result of `HashPayload` to `SigningPayload` variables anyway, I think it just creates confusion.